src/test/java/seedu/address/logic/LogicManagerTest.java

```diff
@@ -0,0 +1,510 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import com.google.common.eventbus.Subscribe;
+
+import seedu.address.commons.core.EventsCenter;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.events.model.EntryBookChangedEvent;
+import seedu.address.commons.events.ui.JumpToListRequestEvent;
+import seedu.address.commons.events.ui.ShowHelpRequestEvent;
+import seedu.address.logic.commands.AddCommand;
+import seedu.address.logic.commands.ClearCommand;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.DeleteCommand;
+import seedu.address.logic.commands.ExitCommand;
+import seedu.address.logic.commands.FindCommand;
+import seedu.address.logic.commands.HelpCommand;
+import seedu.address.logic.commands.HistoryCommand;
+import seedu.address.logic.commands.ListCommand;
+import seedu.address.logic.commands.SelectCommand;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.EntryBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.ReadOnlyEntryBook;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.entry.Entry;
+import seedu.address.model.entry.Name;
+import seedu.address.model.tag.Tag;
+import seedu.address.testutil.EntryBuilder;
+
+
+public class LogicManagerTest {
+
+    /**
+     * See https://github.com/junit-team/junit4/wiki/rules#temporaryfolder-rule
+     */
+    @Rule
+    public TemporaryFolder saveFolder = new TemporaryFolder();
+
+    private Model model;
+    private Logic logic;
+
+    //These are for checking the correctness of the events raised
+    private ReadOnlyEntryBook latestSavedEntryBook;
+    private boolean helpShown;
+    private Index targetedJumpIndex;
+
+    @Subscribe
+    private void handleLocalModelChangedEvent(EntryBookChangedEvent abce) {
+        latestSavedEntryBook = new EntryBook(abce.data);
+    }
+
+    @Subscribe
+    private void handleShowHelpRequestEvent(ShowHelpRequestEvent she) {
+        helpShown = true;
+    }
+
+    @Subscribe
+    private void handleJumpToListRequestEvent(JumpToListRequestEvent je) {
+        targetedJumpIndex = Index.fromZeroBased(je.targetIndex);
+    }
+
+    @Before
+    public void setUp() {
+        model = new ModelManager();
+        logic = new LogicManager(model);
+        EventsCenter.getInstance().registerHandler(this);
+
+        latestSavedEntryBook = new EntryBook(model.getEntryBook()); // last saved assumed to be up to date
+        helpShown = false;
+        targetedJumpIndex = null;
+    }
+
+    @After
+    public void tearDown() {
+        EventsCenter.clearSubscribers();
+    }
+
+    @Test
```

Comment:
We can probably add an issue for V0.3 to tidy up the names of the tests from the provided ones:
`logicManager_invalidCommand_exception` compared to `execute_invalid` for example